python-frontend/src/main/java/org/sonar/python/semantic/SymbolImpl.java

```diff
@@ -116,4 +118,19 @@ void setType(@Nullable Type type) {
   public Type type() {
     return type;
   }
+
+  /*
+   * 'returnType' only applies to function symbols.
+   * It's not defined in FunctionSymbolImpl class because FunctionSymbolImpl must always have a definition location.
+   * Rules implemented in MMF-1845 need to know types of external libraries and symbols coming from said external libraries
+   * don't have a definition location.
```

Comment:
I am not sure I understand why we can't have the `returnType` in `FunctionSymbolImpl` nevertheless?
Isn't it possible to make the definition location optional, or have a default value for it?